In [1]:
import pandas as pd
import json

df = pd.read_csv("C:/Users/kevin/ETL/Postgres/workshop_001/candidates.csv", delimiter=';')

# Exploratory data analysis

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   First Name                 50000 non-null  object
 1   Last Name                  50000 non-null  object
 2   Email                      50000 non-null  object
 3   Application Date           50000 non-null  object
 4   Country                    50000 non-null  object
 5   YOE                        50000 non-null  int64 
 6   Seniority                  50000 non-null  object
 7   Technology                 50000 non-null  object
 8   Code Challenge Score       50000 non-null  int64 
 9   Technical Interview Score  50000 non-null  int64 
dtypes: int64(3), object(7)
memory usage: 3.8+ MB


Podemos ver que el csv dispone de 10 columnas el cual, el tipo de los datos de la mayoria son de texto o varchar, mientras que YOE, Code Challenge Score y Technical Interview Score son valores tipo enteros

In [21]:
df.head()

First Name   Last Name                      Email Application Date  \
0  Bernadette   Langworth        leonard91@yahoo.com       2021-02-26   
1      Camryn    Reynolds        zelda56@hotmail.com       2021-09-09   
2       Larue      Spinka   okey_schultz41@gmail.com       2020-04-14   
3        Arch      Spinka     elvera_kulas@yahoo.com       2020-10-01   
4       Larue  Altenwerth  minnie.gislason@gmail.com       2020-05-20   

   Country  YOE  Seniority                         Technology  \
0   Norway    2     Intern                      Data Engineer   
1   Panama   10     Intern                      Data Engineer   
2  Belarus    4  Mid-Level                     Client Success   
3  Eritrea   25    Trainee                          QA Manual   
4  Myanmar   13  Mid-Level  Social Media Community Management   

   Code Challenge Score  Technical Interview Score  
0                     3                          3  
1                     2                         10  
2                    10                          9  
3                     7                          1  
4                     9                          7

Se hace una visualizacion a como estan organizados los datos dentro de la tabla

In [15]:
print(df.duplicated().sum())

0


Para los valores duplicados da un 0 es decir no hay filas duplicadas 

In [16]:
df["Seniority"].value_counts()

Intern       7255
Mid-Level    7253
Trainee      7183
Junior       7100
Architect    7079
Lead         7071
Senior       7059
Name: Seniority, dtype: int64

En este caso quise hacer el conteo de la columna seniority debido a que hay gran variedad de niveles de experiencia laborales, pero todos se encuentran dentro de un rango de 7000 en adelante

In [17]:
df.isnull().sum()

First Name                   0
Last Name                    0
Email                        0
Application Date             0
Country                      0
YOE                          0
Seniority                    0
Technology                   0
Code Challenge Score         0
Technical Interview Score    0
dtype: int64

Aqui rectificamos que no hayan valores nulos, y efectivamente no hay redundancia en ninguno de los datos

In [18]:
df.describe()

YOE  Code Challenge Score  Technical Interview Score
count  50000.000000          50000.000000               50000.000000
mean      15.286980              4.996400                   5.003880
std        8.830652              3.166896                   3.165082
min        0.000000              0.000000                   0.000000
25%        8.000000              2.000000                   2.000000
50%       15.000000              5.000000                   5.000000
75%       23.000000              8.000000                   8.000000
max       30.000000             10.000000                  10.000000

En este caso se verifico un analisis de los valores de las columnas que son cuantitativos, en donde en cada columna hay 50.000 observaciones, se puede ver que el YOE(years of experience) cuenta con una media de 15.28 asi que en promedio, los candidatos tienen alrededor de 15.29 años de experiencia laboral. Por otra parte la desviacion estandar de YOE es la que contiene un valor mas alto es decir que hay mas variabilidad, el minimo de las 3 tiene un 0 por lo tanto hay candidatos que no cuentan co eperiencia laboral, y en los score sacaron la peor puntuacion, pero tambien hay candidatos excepcionales que en como maximo obtuvieron los mejores score osea 10.

In [19]:
df.nunique()

First Name                    3007
Last Name                      474
Email                        49833
Application Date              1646
Country                        244
YOE                             31
Seniority                        7
Technology                      24
Code Challenge Score            11
Technical Interview Score       11
dtype: int64

En este caso verificamos la cantidad de valores unicos y se puede notar que, hay nombres y apellidos que se repiten, en el caso de los emails no todos son diferentes pero si la mayoria, hay candidatos de 244 paises en total, con 7 tecnologias, para los años de experiencia laboras hay candidatos que van desde 0 hasta los 31 años y lo demas hace referencia de los score que tiene sentido porque la escala va de 0 a 10

In [4]:
df['Technology'].unique()

array(['Data Engineer', 'Client Success', 'QA Manual',
       'Social Media Community Management', 'Adobe Experience Manager',
       'Sales', 'Mulesoft', 'DevOps', 'Development - CMS Backend',
       'Salesforce', 'System Administration', 'Security',
       'Game Development', 'Development - CMS Frontend',
       'Security Compliance', 'Development - Backend', 'Design',
       'Business Analytics / Project Management',
       'Development - Frontend', 'Development - FullStack',
       'Business Intelligence', 'Database Administration',
       'QA Automation', 'Technical Writing'], dtype=object)